src/main/java/io/supertokens/webserver/api/bulkimport/AddBulkImportUsers.java

```diff
@@ -0,0 +1,124 @@
+import io.supertokens.pluginInterface.bulkimport.exceptions.InvalidBulkImportDataException;
+import io.supertokens.pluginInterface.multitenancy.AppIdentifier;
+import io.supertokens.pluginInterface.multitenancy.AppIdentifierWithStorage;
+import io.supertokens.pluginInterface.multitenancy.TenantConfig;
+import io.supertokens.webserver.InputParser;
+import io.supertokens.webserver.WebserverAPI;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonPrimitive;
+
+public class AddBulkImportUsers extends WebserverAPI {
+    private static final int MAX_USERS_TO_ADD = 10000;
+
+    public AddBulkImportUsers(Main main) {
+        super(main, "");
+    }
+
+    @Override
+    public String getPath() {
+        return "/bulk-import/add-users";
+    }
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+        JsonObject input = InputParser.parseJsonObjectOrThrowError(req);
+        JsonArray users = InputParser.parseArrayOrThrowError(input, "users", false);
+
+        if (users.size() > MAX_USERS_TO_ADD) {
+            JsonObject errorResponseJson = new JsonObject();
+            errorResponseJson.addProperty("error", "You can only add 1000 users at a time.");
+            super.sendJsonResponse(400, errorResponseJson, resp);
+            return;
+        }
+
+        AppIdentifier appIdentifier = null;
+        try {
+            appIdentifier = getTenantIdentifierFromRequest(req).toAppIdentifier();
+        } catch (ServletException e) {
+            throw new ServletException(e);
+        }
+
+        TenantConfig[] allTenantConfigs = Multitenancy.getAllTenantsForApp(appIdentifier, main);
+        ArrayList<String> validTenantIds = new ArrayList<>();
+        Arrays.stream(allTenantConfigs)
+                .forEach(tenantConfig -> validTenantIds.add(tenantConfig.tenantIdentifier.getTenantId()));
+
+        JsonArray errorsJson = new JsonArray();
+        ArrayList<BulkImportUser> usersToAdd = new ArrayList<>();
+
+        for (int i = 0; i < users.size(); i++) {
+            try {
+                usersToAdd.add(new BulkImportUser(users.get(i).getAsJsonObject(), validTenantIds, null));
+            } catch (InvalidBulkImportDataException e) {
+                JsonObject errorObj = new JsonObject();
+
+                JsonArray errors = e.errors.stream()
+                        .map(JsonPrimitive::new)
+                        .collect(JsonArray::new, JsonArray::add, JsonArray::addAll);
+
+                errorObj.addProperty("index", i);
+                errorObj.add("errors", errors);
+
+                errorsJson.add(errorObj);
+            } catch (Exception e) {
```

Comment:
catching all exceptions is a bad idea.. see how other APis are structured

Reply:
Fixed it.